crates/ra_ide/src/completion.rs

```diff
@@ -65,23 +65,20 @@ pub(crate) fn completions(
     let ctx = CompletionContext::new(db, position, config)?;
 
     let mut acc = Completions::default();
-    if ctx.attribute_under_caret.is_some() {
```

Comment:
Deleting this, we get more false-positive completions in attribute paths: 
![image](https://user-images.githubusercontent.com/2690773/80974045-de70ec80-8e28-11ea-896a-d11692d3676c.png)
![image](https://user-images.githubusercontent.com/2690773/80974059-e335a080-8e28-11ea-9c1e-10e79f1c7dc0.png)

Any any other completion besides `complete_attribute::complete_attribute` one help with completion in attributes now?

Reply:
Pushed an explicit fix for this, though I know wonder if we actually should allow this? You can use proc-macros with paths, right? like `#[async_std::main]`?

Reply:
As far as I understand it, we cannot really know what an attribute input will be required to be :)

For instance, Serde also requires to specify a path, but in a different way: https://serde.rs/container-attrs.html#default--path 

We might want to propose to complete the macro-specific things on top, before all pathes.

Reply:
I mean, in `#[async_std::main]` nothing is an attribute input. It's just a usual path, which obeys usual name resolution rules. Derive macros are special (I think?) but attr like macros are not. the following works:

```
#[ra_db::salsa::query_group(HirDatabaseStorage)]
#[ra_db::salsa::requires(salsa::Database)]
pub trait HirDatabase: DefDatabase + Upcast<dyn DefDatabase> {

```

Reply:
Oh joy, so we mkight
So, in theory, we can parse it better without a custom parsing logic I've submitted recently.

Reply:
Oh joy, so we might need some completion there after all, but with special checks to filter out things like macros?

Does it also mean that we can parse the attrributes better without [a custom parsing logic](https://github.com/rust-analyzer/rust-analyzer/blob/master/crates/ra_ide/src/completion/complete_attribute.rs#L180) I've submitted recently?

Reply:
Another option here is make `get_derive_names_in_scope` search for proc-macro recursively?